closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -185,24 +201,26 @@ protected Key engineDoPhase(Key key, boolean lastPhase)
         // Validate public key.
         validate(privateKeyOps, ecKey);
 
-        if (ecKey instanceof ECPublicKeyImpl keyImpl) {
-            this.publicKey = keyImpl;
-
-            int keyLenBits = this.publicKey.getParams().getCurve().getField().getFieldSize();
-            this.secretLen = (keyLenBits + 7) >> 3;
-
-            return null;
-        } else {
-            boolean absent = NativeECUtil.putCurveIfAbsent("ECKeyImpl", Boolean.FALSE);
-            /* only print the first time a curve is used */
-            if (absent && nativeCryptTrace) {
-                System.err.println("Only ECPrivateKeyImpl and ECPublicKeyImpl" +
-                        " are supported by the native implementation, " +
-                        "using Java crypto implementation for key agreement.");
+        this.publicKey = ecKey;
+        nativePublicKey = NativeECUtil.getPublicKeyNativePtr(this.publicKey);
+        if (nativePublicKey == -1) {
```

Comment:
Please use the local rather than the field, and qualify field references with `this.`.

Reply:
Changed.